opentelemetry/proto/profiles/v1/profiles.proto

```diff
@@ -274,6 +281,9 @@ message Location {
 
   // Reference to an attribute set from the Profile's attribute set table.
   repeated uint32 attribute_set_indices = 5;
+
+  // Type of frame (e.g. kernel, native, python, hotspot, php). Index into string table.
+  uint32 type_index = 5;
```

Comment:
This will be useful when profiling mixed workloads (e.g. Python calling into C). Allows
distinguishing between different runtimes. It is not an enumerated type to allow for
easy extensibility.

Reply:
Feedback from the committee meeting: A cheaper way of representing the type may be a special mapping type for each runtime?